ToolBox/IQFeed/IQFeedFileHistoryProvider.cs

```diff
@@ -0,0 +1,207 @@
+        /// <returns>Converted Tick</returns>
+        private IEnumerable<BaseData> GetDataFromTickMessages(string filename, HistoryRequest request, bool isEquity)
+        {
+            // We need to discard ticks which are not impacting the price, i.e those having BasisForLast = O
+            // To get a better understanding how IQFeed is resampling ticks, have a look to this algorithm:
+            // https://github.com/mathpaquette/IQFeed.CSharpApiClient/blob/1b33250e057dfd6cd77e5ee35fa16aebfc8fbe79/src/IQFeed.CSharpApiClient.Extensions/Lookup/Historical/Resample/TickMessageExtensions.cs#L41
+            foreach (var tick in TickMessage.ParseFromFile(filename).Where(t => t.BasisForLast != 'O'))
+            {
+                var time = isEquity ? tick.Timestamp : tick.Timestamp.ConvertTo(TimeZones.NewYork, TimeZones.EasternStandard);
+                yield return new Tick(time, request.Symbol, (decimal)tick.Last, (decimal)tick.Bid, (decimal)tick.Ask)
+                { Quantity = tick.LastSize };
+            }
+
+            File.Delete(filename);
+        }
+
+        /// <summary>
+        /// Stream IQFeed DailyWeeklyMonthlyMessage from disk to Lean TradeBar
+        /// </summary>
+        /// <param name="filename"></param>
+        /// <param name="request"></param>
+        /// <param name="isEquity"></param>
+        /// <returns>Converted TradeBar</returns>
+        private IEnumerable<BaseData> GetDataFromDailyMessages(string filename, HistoryRequest request, bool isEquity)
+        {
+            foreach (var daily in DailyWeeklyMonthlyMessage.ParseFromFile(filename))
+            {
+                var dstartTime = daily.Timestamp.Date;
+                if (!isEquity) dstartTime = dstartTime.ConvertTo(TimeZones.NewYork, TimeZones.EasternStandard);
+                yield return new TradeBar(
+                    dstartTime,
+                    request.Symbol,
+                    (decimal)daily.Open,
+                    (decimal)daily.High,
+                    (decimal)daily.Low,
+                    (decimal)daily.Close,
+                    daily.PeriodVolume,
+                    request.Resolution.ToTimeSpan()
+                );
+            }
+
+            File.Delete(filename);
+        }
+
+        /// <summary>
+        /// Stream IQFeed IntervalMessage from disk to Lean TradeBar
+        /// </summary>
+        /// <param name="filename"></param>
+        /// <param name="request"></param>
+        /// <param name="isEquity"></param>
+        /// <returns>Converted TradeBar</returns>
+        private IEnumerable<BaseData> GetDataFromIntervalMessages(string filename, HistoryRequest request, bool isEquity)
+        {
+            foreach (var interval in IntervalMessage.ParseFromFile(filename))
+            {
+                var istartTime = interval.Timestamp - request.Resolution.ToTimeSpan();
+                if (!isEquity) istartTime = istartTime.ConvertTo(TimeZones.NewYork, TimeZones.EasternStandard);
```

Comment:
This tickers data time zone might not be `TimeZones.EasternStandard`, thinking about the futures for example, would be nice to fetch the entry from the MarketHoursDataBase. Same comment for other time zone conversions.

Reply:
might be a good future improvement but this code was basically to improve the performance :)